### ISSUE TITLE
LSTM overfitting?

### ISSUE THREAD

**Issue by tsly123** [keras-team/keras#9024]
Hi,
I am trying to replicate the result from gesture-based emotion recognition experiment in this [paper](https://www.sciencedirect.com/science/article/pii/S0893608015001847) with same raw video dataset.  In the paper, they employed 3D CNN while I am using 10-fold cross-validation with 2 different approaches (due to the limited dataset - less than 300 video): 
- CNN + LSTM
- Features extracted by InceptionV3 and then fed to LSTM.
The task is classifying the sequences of frames into 11 emotion labels.

No matter how I tried, the accuracy only reached 30% for both approaches (paper: 50% test set with 30 samples). In addition, the val_acc very slowly increased from below 5% to max 30% (spike) at over 200 epochs and then decreased. The val_losses increased as epochs increased. 
I have tried the followed changing but the result is still same:
- model parameters: number of layers, varied capacity, number of LSTM units, learning rate, optimizers, frames/sequence (10 and 4), etc.
- Augmented the frames in the dataset.
- different layer in InceptionV3 for extracting features.
- grayscale and Sobel (edge detector) dataset, the accuracies were almost same. 

Any suggestion to increase accuracy?
Thank you.

`for train_index, test_index in kfold:

    model = Sequential() 
    
    model.add(TimeDistributed(Conv2D(32, (7, 7), strides=(2, 2),
        activation='relu', padding='same'), input_shape=(seq_samples, height, width, 3)))
    model.add(TimeDistributed(Conv2D(32, (3,3),
        kernel_initializer="he_normal", activation='relu')))
    model.add(BatchNormalization())
    model.add(TimeDistributed(MaxPooling2D((2, 2), strides=(2, 2))))

    model.add(TimeDistributed(Conv2D(64, (3,3),
        padding='same', activation='relu')))
    model.add(TimeDistributed(Conv2D(64, (3,3),
        padding='same', activation='relu')))
    model.add(BatchNormalization())
    model.add(TimeDistributed(MaxPooling2D((2, 2), strides=(2, 2))))

    model.add(TimeDistributed(Conv2D(64, (3,3),
        padding='same', activation='relu')))
    model.add(TimeDistributed(Conv2D(64, (3,3),
        padding='same', activation='relu')))
    model.add(BatchNormalization())
    model.add(TimeDistributed(MaxPooling2D((2, 2), strides=(2, 2))))

    model.add(TimeDistributed(Flatten()))

    model.add(LSTM(256, return_sequences=False))
    model.add(Dropout(0.5))
    model.add(Dense(11,activity_regularizer=regularizers.l1_l2(0.001), activation='softmax'))

    model.summary()

    sgd = SGD(lr=0.001)

    model.compile(loss='categorical_crossentropy',
                  optimizer=sgd,
                  metrics=['accuracy'])

        ###################
    os.mkdir(callbackpath + '\%s' %k)
    subcallback = os.path.abspath(callbackpath + '\%s' %k)
    k = k + 1
    
    checkmodel = ModelCheckpoint(filepath='%s\{val_acc:.2f}-{epoch:02d}-10fol-LSTM.h5' % subcallback,
                                   monitor = 'val_acc',
                                   verbose=0,
                                   save_best_only=True,
                                   mode = 'max')

    checkweight = ModelCheckpoint(filepath='%s\{val_acc:.2f}-{epoch:02d}-10fol-LSTM.hdf5' % subcallback,
                                   monitor = 'val_acc',
                                   verbose=0,
                                   save_best_only=True,
                                   mode = 'max')

    training = model.fit(x_train[train_index], y_train1[train_index],
                         batch_size = 128,
                         epochs = 1000,
                         verbose = 1,
                         class_weight = class_weight_for_training,   #sklearn.utils  class_weight for imbalance data
                         validation_data=(x_train[test_index], y_train1[test_index]),
                         shuffle = True,
                         callbacks= [checkmodel, checkweight])

    acc = model.evaluate(x_train[test_index], y_train1[test_index],
                         verbose = 1)

    K.clear_session()`

A typical training:
Epoch 1/1000
6003/6003 [==============================] - 36s 6ms/step - loss: 2.4660 - acc: 0.0816 - val_loss: 2.3885 - val_acc: 0.1437
Epoch 2/1000
6003/6003 [==============================] - 16s 3ms/step - loss: 2.4488 - acc: 0.0900 - val_loss: 2.3917 - val_acc: 0.1257
Epoch 3/1000
6003/6003 [==============================] - 14s 2ms/step - loss: 2.4411 - acc: 0.0906 - val_loss: 2.3970 - val_acc: 0.1018
Epoch 4/1000
6003/6003 [==============================] - 14s 2ms/step - loss: 2.4321 - acc: 0.0866 - val_loss: 2.3991 - val_acc: 0.0928
Epoch 5/1000
6003/6003 [==============================] - 17s 3ms/step - loss: 2.4222 - acc: 0.0903 - val_loss: 2.3992 - val_acc: 0.1003
....
Epoch 100/1000
6003/6003 [==============================] - 15s 3ms/step - loss: 1.6579 - acc: 0.3848 - val_loss: 2.4183 - val_acc: 0.0943
Epoch 101/1000
6003/6003 [==============================] - 14s 2ms/step - loss: 1.6551 - acc: 0.3870 - val_loss: 2.4136 - val_acc: 0.1018
Epoch 102/1000
6003/6003 [==============================] - 14s 2ms/step - loss: 1.6224 - acc: 0.3951 - val_loss: 2.4257 - val_acc: 0.1003
Epoch 103/1000
6003/6003 [==============================] - 13s 2ms/step - loss: 1.6196 - acc: 0.3978 - val_loss: 2.4426 - val_acc: 0.0913
Epoch 104/1000
6003/6003 [==============================] - 13s 2ms/step - loss: 1.6029 - acc: 0.3958 - val_loss: 2.4291 - val_acc: 0.0973
Epoch 105/1000
6003/6003 [==============================] - 14s 2ms/step - loss: 1.5826 - acc: 0.4083 - val_loss: 2.4287 - val_acc: 0.1048
.....
Epoch 280/1000
6003/6003 [==============================] - 16s 3ms/step - loss: 0.1454 - acc: 0.9589 - val_loss: 2.9661 - val_acc: 0.2036
Epoch 281/1000
6003/6003 [==============================] - 16s 3ms/step - loss: 0.1427 - acc: 0.9609 - val_loss: 2.9746 - val_acc: 0.1961
Epoch 282/1000
6003/6003 [==============================] - 15s 3ms/step - loss: 0.1407 - acc: 0.9612 - val_loss: 2.9613 - val_acc: 0.2021
Epoch 283/1000
6003/6003 [==============================] - 14s 2ms/step - loss: 0.1404 - acc: 0.9624 - val_loss: 2.9734 - val_acc: 0.1976
Epoch 284/1000
6003/6003 [==============================] - 15s 2ms/step - loss: 0.1367 - acc: 0.9632 - val_loss: 2.9716 - val_acc: 0.2066
Epoch 285/1000
6003/6003 [==============================] - 15s 2ms/step - loss: 0.1367 - acc: 0.9649 - val_loss: 2.9724 - val_acc: 0.1976

![capture](https://user-images.githubusercontent.com/35265385/34731359-75fa3760-f5a5-11e7-9041-1b0ae73477ba.JPG)
![capture jpg1](https://user-images.githubusercontent.com/35265385/34731364-79c9dcb0-f5a5-11e7-8594-af8948d317df.JPG)
